libnetwork/networkdb/networkdb.go

```diff
@@ -429,8 +429,11 @@ type TableElem struct {
 // GetTableByNetwork walks the networkdb by the give table and network id and
 // returns a map of keys and values
 func (nDB *NetworkDB) GetTableByNetwork(tname, nid string) map[string]*TableElem {
+	nDB.RLock()
+	root := nDB.indexes[byNetwork].Root()
+	nDB.RUnlock()
```

Comment:
Interesting; so we have a read-lock here, but there's no lock in 2 places I found that mutate this?

https://github.com/moby/moby/blob/d71afd73bf54599a2d2aac4e5072719bc62adcb8/libnetwork/networkdb/networkdb.go#L761-L763

https://github.com/moby/moby/blob/d71afd73bf54599a2d2aac4e5072719bc62adcb8/libnetwork/networkdb/networkdb.go#L776-L778

Reply:
Ah, ignore me; from https://github.com/moby/moby/pull/44501

> The major changes compared to the old radix are when modifying (insert or delete) a tree, and those are pretty self-contained: we replace the entire immutable tree under a lock.

Looks like caller is acquiring a lock indeed. although there's some odd bits because it's doing the swap inside of a `nDB.indexes[byNetwork].Root().WalkPrefix([]byte("/"+nid),` 🤔 ..  I recall now that there was some weird stuff in that area that originally caused some panics the "mutable" radix, well 🤷